aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,10 +1,171 @@
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/package.json
+  :language: json
+  :linenos:
+  :emphasize-lines: 6
+
+Run ``npm install`` to then install the specified packages.
+
+.. note:: Visual Studio hides the package.json file by default. To edit package.json right-click on the npm node under Dependencies in the solution explorer and select **Open package.json**. Visual Studio will automatically install npm packages whenever package.json is modified.
+
+In your ``gulpfile.js`` import the ``gulp-concat`` module:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/gulpfile.js
+  :language: js
+  :linenos:
+  :lines: 4-8
+  :emphasize-lines: 3
+
+Use globbing patterns to specify the files that you want to bundle and minify:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/gulpfile.js
+  :language: js
+  :linenos:
+  :lines: 14-19
+
+You can then define gulp tasks that run ``concat`` on the desired files and output the result to your webroot:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/gulpfile.js
+  :language: js
+  :linenos:
+  :lines: 31-43
+  :emphasize-lines: 3, 10
+
+Minification
+------------
+
+Minification performs a variety of different code optimizations to reduce the size of requested assets (such as CSS, image, JavaScript files). Common results of minification include removing unnecessary white space and comments, and shortening variable names to one character. 
+
+Consider the following JavaScript function:
+
+.. code-block:: javascript
+
+	AddAltToImg = function (imageTagAndImageID, imageContext) {
+		///<signature>
+		///<summary> Adds an alt tab to the image
+		// </summary>
+		//<param name="imgElement" type="String">The image selector.</param>
+		//<param name="ContextForImage" type="String">The image context.</param>
+		///</signature>
+		var imageElement = $(imageTagAndImageID, imageContext);
+		imageElement.attr('alt', imageElement.attr('id').replace(/ID/, ''));
+	}
+
+After minification, the function can be reduced to the following:
+
+.. code-block:: javascript
+
+	AddAltToImg=function(t,a){var r=$(t,a);r.attr("alt",r.attr("id").replace(/ID/,""))};
+	
+In addition to removing the comments and unnecessary whitespace, the following parameters and variable names were renamed (shortened) as follows:
+
+==================  =======  
+Original            Renamed      
+==================  =======  
+imageTagAndImageID  t  
+imageContext        a  
+imageElement        r   
+==================  =======  
+
+To minify your JavaScript files you can use the `gulp-uglify <https://www.npmjs.com/package/gulp-uglify>`__ task. For CSS you can use the `gulp-cssmin <https://www.npmjs.com/package/gulp-cssmin>`__ task. Install these packages using npm as before:
```

Comment:
I'm ok either way. I believe this will localize ok and it's grammatically correct (yes it is legal to end in a preposition).